### PR TITLE
[libmagic] Explicitly toggle features.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Libmagic supports some compression libraries that by default are enabled unpredictably, if they are found in the system. This PR adds features to the port that enable support for these compression libraries, and makes sure the features are disabled if not specified.

Validated by building the port with all features enabled on Windows and Linux, and importing it in a project with static library linkage on Windows.

### CODE REVIEW

File: ports/libmagic/portfile.cmake [hunk 1]
```diff
@@ -31,9 +32,25 @@ if(VCPKG_TARGET_IS_WINDOWS)
     set(VCPKG_CXX_FLAGS "${VCPKG_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_WARNINGS")
 endif()
 
+set(CONFIG_OPTIONS)
+
+foreach(feature zlib bzlib xzlib zstdlib)
```
Comment (FrankXie05) [inline ports/libmagic/portfile.cmake:37]:
```suggestion
set(FEATURE_OPTIONS)

foreach(feature zlib bzlib xzlib zstdlib)
```
Are these all features changes?

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/libmagic/portfile.cmake (hunk 1 above)
Comment (dg0yt) [inline ports/libmagic/portfile.cmake:37]:
```suggestion
foreach(feature IN ITEMS zlib bzlib xzlib zstdlib)
```

TBH I dislike most feature names. They match neither cababilities (preferred) nor port names (seen quite often).

Comment (teo-tsirpanis) [reply ports/libmagic/portfile.cmake]: Renamed features to match the ports they depend on.

Comment (teo-tsirpanis) [reply ports/libmagic/portfile.cmake]: Renamed.

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
All features are tested successfully in the following triplet:

- x86-windows
- x64-windows
- x64-windows-stataic

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**